src/activity_handlers.h

```diff
@@ -53,7 +53,54 @@ enum do_activity_reason : int {
     ALREADY_WORKING,        // somebody is already working there
     NEEDS_VEH_DECONST,       // There is a vehicle part there that we can deconstruct, given the right tools.
     NEEDS_VEH_REPAIR,       // There is a vehicle part there that can be repaired, given the right tools.
-    NEEDS_FISHING           // This spot can be fished, if the right tool is present.
+    NEEDS_FISHING,           // This spot can be fished, if the right tool is present.
+    NUM_DO_ACTIVITY_REASON
+};
+
+namespace io
+{
+
+template<>
+std::string enum_to_string<do_activity_reason>( const do_activity_reason act )
```

Comment:
I'm not sure if it's the reason of the linker errors, but these definitions of `enum_to_string` can go into the cpp files instead of being in headers.

Reply:
Yep, with thta and some changes to a default constructor, it worked, thanks.